pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -97,8 +98,24 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
 		targetName = ar.Request.Name
 	}
 
+	modifiedDataVolume := dataVolume.DeepCopy()
+	modified := false
+
+	config, _ := wh.cdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+	if config != nil && config.Spec.DataVolumeTTLSeconds != nil &&
+		modifiedDataVolume.Annotations[controller.AnnDeleteAfterCompletion] != "false" {
```

Comment:
`.Annotations` can be nil here/not have that key in it, no?

Reply:
Reading from a nil map is allowed. In both cases it will get empty string, which is good for us so we can add the annotation with true.

Reply:
right